iocore/net/SSLCertLookup.cc

```diff
@@ -45,26 +47,19 @@
 #endif /* SSL_CTX_set_tlsext_ticket_key_cb */
 
 struct SSLAddressLookupKey {
-  explicit SSLAddressLookupKey(const IpEndpoint &ip) : sep(0)
+  explicit SSLAddressLookupKey(const IpEndpoint &ip)
   {
-    static const char hextab[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
-
-    int nbytes;
-    uint16_t port = ntohs(ip.port());
-
-    // For IP addresses, the cache key is the hex address with the port concatenated. This makes the lookup
-    // insensitive to address formatting and also allow the longest match semantic to produce different matches
-    // if there is a certificate on the port.
-    nbytes = ats_ip_to_hex(&ip.sa, key, sizeof(key));
-    if (port) {
-      sep           = nbytes;
-      key[nbytes++] = '.';
-      key[nbytes++] = hextab[(port >> 12) & 0x000F];
-      key[nbytes++] = hextab[(port >> 8) & 0x000F];
-      key[nbytes++] = hextab[(port >> 4) & 0x000F];
-      key[nbytes++] = hextab[(port)&0x000F];
+    // For IP addresses, the cache key is the hex address with the port concatenated. This makes the
+    // lookup insensitive to address formatting and also allow the longest match semantic to produce
+    // different matches if there is a certificate on the port.
+
+    ts::FixedBufferWriter w{key, sizeof(key)};
+    w.print("{}", ts::bwf::As_Hex(ip));
```

Comment:
I'm a little confused here. I thought we said we'd python formatting? There has to be something in Python's string format to convert an integer to a hex? Like, {:04X} or some such ? It just seems we have a lot of {}, which I assume means "any string", and then do the conversion with helper functions? Why is this better than just printf()? :)

Reply:
There is a significant difference between "{:x}" which means "format normally but in hexadecimal" and "As_Hex" which means "dump the raw memory in hexadecimal". In particular, on a little endian machine, "{:x}" vs. "As_Hex" will yield reversed results for integral types. I've already changed the name of this once or twice to accomodate previous comments and selected "As_Hex" based on what feedback I could extract. Perhaps we could come to some concensus about what a good name is? "HexDump"?

Reply:
Changed to `Hex_Dump`.